diff --git a/node_modules/@emeris/mapper/lib/implementations/cosmos/osmosis/index.js b/node_modules/@emeris/mapper/lib/implementations/cosmos/osmosis/index.js
index 96b2d0c..b2123a5 100644
--- a/node_modules/@emeris/mapper/lib/implementations/cosmos/osmosis/index.js
+++ b/node_modules/@emeris/mapper/lib/implementations/cosmos/osmosis/index.js
@@ -2,17 +2,22 @@
 Object.defineProperty(exports, "__esModule", { value: true });
 const index_1 = require("../index");
 class OsmosisAminoMessageMapper extends index_1.default {
-    swap(transaction, signing_address) {
+  swap(transactions, signing_address) {
+    let routes = [];
+    for (let i = 0; i < transactions.length; i++) {
+        const transaction = transactions[i];
         const pool = transaction.pool;
-        return {
-            type: "osmosis/gamm/swap-exact-amount-in",
-            value: {
-                sender: signing_address,
-                tokenIn: transaction.from,
-                tokenOutMinAmount: transaction.to.amount,
-                routes: [{ poolId: pool.id, tokenOutDenom: transaction.to.denom }]
-            }
-        };
+        routes.push({ poolId: pool.id, tokenOutDenom: transaction.to.denom });
     }
+    return {
+        type: "osmosis/gamm/swap-exact-amount-in",
+        value: {
+            sender: signing_address,
+            tokenIn: transactions[0].from,
+            tokenOutMinAmount: transactions[transactions.length-1].to.amount,
+            routes
+        }
+    }
+  }
 }
 exports.default = OsmosisAminoMessageMapper;
